File: app/src/main/java/com/wadektech/eventshub/database/ConcertsDao.java
package com.wadektech.eventshub.database;


import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import com.wadektech.eventshub.models.Concerts;
import java.util.List;

@Dao
public interface ConcertsDao {

    @Query("SELECT * FROM Concerts ORDER BY date")
    LiveData<List<Concerts>> getAllConcertsAndTheatres();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void saveAllConcertsAndTheatres(List<Concerts> concertsList);
}

File: app/src/main/java/com/wadektech/eventshub/database/ConcertsDao.kt
package com.wadektech.eventshub.database

import androidx.lifecycle.LiveData
import androidx.paging.DataSource
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.wadektech.eventshub.models.Concerts

@Dao
interface ConcertsDao {
    @Query("SELECT * FROM Concerts ORDER BY date")
    fun getAllConcertsAndTheatres(): DataSource.Factory<Int, Concerts>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun saveAllConcertsAndTheatres(concertsList: MutableList<Concerts>)
}